Tasks/AppCenterDistributeV1/appcenterdistribute.ts

```diff
@@ -173,6 +173,10 @@ function publishRelease(apiServer: string, releaseUrl: string, releaseNotes: str
 
     // Builds started by App Center has the commit message set when distribution is enabled
     const commitMessage = process.env['LASTCOMMITMESSAGE'];
+    // Updating the internal_request_source to distinguish the AppCenter triggered build and custom build
+    if(!!commitMessage) {
+        headers["internal-request-source"] = "VSTS_APPCENTER";
```

Comment:
This is not reliable. Plus, if you are using commit message to determine this, you can check the same on distribution backend, because commit message is sent to the release API

Reply:
@DennisPan If we distribute a build manually from UI, that also sends the commit message to distribution patch api, in which case the request-resource would be `Portal`.
Hence we cant rely on checking at distribution backend.

The only other way is to introduce a new variable something like `isTriggeredByAppCenter` and access like `tl.getVariable("isTriggeredByAppCenter")`. This needs some changes at build service level as well.

What do u suggest.

Reply:
Yeah @DennisPan.
But the problem statement is this vsts distribute task can be leveraged in any build definition,
And we want to differentiate if this is triggering by AppCenter for our internal logging.
So AppCenter triggered distribution has internal_request_source as VSTS-AppCenter and the custom triggers has just VSTS.

Reply:
I guess you can use a combination of the existing internal request source and the existence of the commit message from distribution backend?

Reply:
yeah that can be done. but i feel it creates the dependency of checking the literal string `VSTS` at distribution level. In future, if there is any changes to request-source here, then we need to handle it at distribution level.
Do we see any issue adding over here.

Reply:
I don't like the idea of adding a fork of the internal request source in this task. You are right that distribution backend becomes vulnerable to the potential `VSTS` string change. But the header you added here is yet another string that can potentially change. 

Maybe others can chime in here. Can you check with Drew from distribution team?

Reply:
> But the header you added here is yet another string that can potentially change.
Even if the request source changes in this task in future, there would be no other component that gets impacted.

Sure will check with izik/drew from Distribution team on this.